src/support/Dhis2.js

```diff
@@ -312,8 +312,8 @@ class Dhis2 {
       orgUnits: orgUnits,
       period: period,
       quarterPeriod: period,
-      quarterPeriods: DatePeriods.split(period, "quarterly"),
-      monthlyPeriods: DatePeriods.split(period, "monthly"),
+      quarterPeriods: invoiceType.previousPeriods !== undefined ? DatePeriods.split(DatePeriods.previousQuarterByPeriodNumber(period, invoiceType.previousPeriods), "quarterly") : DatePeriods.split(period, "quarterly"),
```

Comment:
I would extract quarterPeriods and monthlyPeriods
in a variable and leave the  return as it was before.

```
  let quarterPeriods = DatePeriods.split(period, "quarterly")
  let monthlyPeriods = DatePeriods.split(period, "monthly")
  if (invoiceType.previousPeriods) {
    quarterPeriods = quarterPeriods.concat(DatePeriods.previousPeriods(DatePeriods.split(period, "quarterly")[0], invoiceType.previousPeriods))
    monthlyPeriods = monthlyPeriods.concat(DatePeriods.previousPeriods(DatePeriods.split(period, "monthly")[0], invoiceType.previousPeriods))
  }
       
```

IMO to be consistent you need to add yearlyPeriods too